download.js

```diff
@@ -81,24 +81,26 @@ function downloadPrebuild (opts, cb) {
     log.info('unpacking @', cachedPrebuild)
     pump(fs.createReadStream(cachedPrebuild), zlib.createGunzip(), tfs.extract(opts.path, {readable: true, writable: true}).on('entry', updateName), function (err) {
       if (err) return cb(err)
-      if (!binaryName) return cb(error.invalidArchive())
 
       var resolved
-      try {
-        resolved = path.resolve(opts.path || '.', binaryName)
-      } catch (err) {
-        return cb(err)
-      }
-      log.info('unpack', 'resolved to ' + resolved)
-
-      if (opts.abi === process.versions.modules) {
+      if (binaryName) {
         try {
-          require(resolved)
+          resolved = path.resolve(opts.path || '.', binaryName)
         } catch (err) {
           return cb(err)
         }
-        log.info('unpack', 'required ' + resolved + ' successfully')
+        log.info('unpack', 'resolved to ' + resolved)
+
+        if (opts.abi === process.versions.modules) {
+          try {
+            require(resolved)
+          } catch (err) {
+            return cb(err)
+          }
+          log.info('unpack', 'required ' + resolved + ' successfully')
+        }
       }
+
       cb(null, resolved)
```

Comment:
> If binaryName is falsey we will cb(null, undefined) right? Just making sure :)

Yes, that's it :-)


> ^ I guess that's part of your use case?

Somewhat... I think that `resolved` is a nice feature for compiled modules to don't need to know where they are so they can be required, but in my case since I'm storing binaries I'm setting explicitly where the final files are and instead I've added an `index.js` file that return a string with the binary file path, just as PhantomJS does.

By the way, I'm using this module with nodeos-cross-toolchain module and works like a charm :-D